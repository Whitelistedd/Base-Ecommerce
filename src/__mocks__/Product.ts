export const TEST_PRODUCT1 = {
	categories: ['hoodies'],
	color: ['black'],
	createdAt: '2022-06-16T11:18:10.448Z',
	desc: '',
	gender: ['men'],
	img: [
		'https://i.ibb.co/93ksW04/Untitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp',
	],
	inStock: 8,
	price: 3000,
	size: ['XL', 'L', 'M'],
	title: "MEN'S QUARTER ZIP HOODIE",
	updatedAt: '2022-06-16T11:18:10.448Z',
	__v: 0,
	_id: '62ab1172bb9ddb4cfa932c50',
}
export const TEST_PRODUCT2 = {
	categories: ['hoodies'],
	color: ['black'],
	createdAt: '2022-06-16T11:18:10.448Z',
	desc: '',
	gender: ['men'],
	img: [
		'https://i.ibb.co/93ksW04/Untitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp',
	],
	inStock: 8,
	price: 3000,
	size: ['XL', 'L', 'M'],
	title: "MEN'S QUARTER ZIP HOODIE",
	updatedAt: '2022-06-16T11:18:10.448Z',
	__v: 0,
	_id: '62ab1172bb9ddb4cfa932c51',
}
export const TEST_PRODUCT3 = {
	categories: ['hoodies'],
	color: ['black'],
	createdAt: '2022-06-16T11:18:10.448Z',
	desc: '',
	gender: ['men'],
	img: [
		'https://i.ibb.co/93ksW04/Untitleddesign-d3cc44e5-2d2b-4c6b-a576-1aaa9fa2855f-600x.webp',
	],
	inStock: 8,
	price: 3000,
	size: ['XL', 'L', 'M'],
	title: "MEN'S QUARTER ZIP HOODIE",
	updatedAt: '2022-06-16T11:18:10.448Z',
	__v: 0,
	_id: '62ab1172bb9ddb4cfa932c52',
}
