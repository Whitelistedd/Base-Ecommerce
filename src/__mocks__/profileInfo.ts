const profileInfo = {
  firstName: 'test',
  lastName: 'test',
  address: {
    address: 'testAddress',
    city: 'testCity',
    apartment: 'testApartment',
    zipCode: 'testZipCode',
    country: '',
  },
}

export default profileInfo
