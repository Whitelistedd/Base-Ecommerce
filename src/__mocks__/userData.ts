const userData = {
	name: 'user',
	email: 'user@example.com',
}

export default userData
