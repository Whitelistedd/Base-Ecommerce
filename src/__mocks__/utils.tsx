/* eslint-disable react/display-name */

import * as React from 'react'

import { QueryClient, QueryClientProvider } from '@tanstack/react-query'

import { render } from '@testing-library/react'

const createTestQueryClient = () =>
	new QueryClient({
		defaultOptions: {
			queries: {
				retry: false,
			},
		},
		logger: {
			log: console.log,
			warn: console.warn,
			// eslint-disable-next-line @typescript-eslint/no-empty-function
			error: () => {},
		},
	})

export function renderWithClient(client: QueryClient, ui: React.ReactElement) {
	const { rerender, ...result } = render(
		<QueryClientProvider client={client}>{ui}</QueryClientProvider>
	)
	return {
		...result,
		rerender: (rerenderUi: React.ReactElement) =>
			rerender(
				<QueryClientProvider client={client}>{rerenderUi}</QueryClientProvider>
			),
	}
}

export function createWrapper() {
	const testQueryClient = createTestQueryClient()
	return ({ children }: { children: React.ReactNode }) => (
		<QueryClientProvider client={testQueryClient}>
			{children}
		</QueryClientProvider>
	)
}
