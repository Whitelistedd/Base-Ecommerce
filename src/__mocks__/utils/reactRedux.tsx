import { Provider } from 'react-redux'
import cartSlice from '@/redux/slices/cart'
import { configureStore } from '@reduxjs/toolkit'
import userSlice from '@/redux/slices/user'

export const mockStore = configureStore({
  reducer: {
    user: userSlice,
    cart: cartSlice,
  },
})

export function wrapWithRedux(children: React.ReactNode) {
  return <Provider store={mockStore}>{children}</Provider>
}
