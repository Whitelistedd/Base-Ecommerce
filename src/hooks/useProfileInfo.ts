import { ProfileInfo } from 'types/GlobalTypes.model'
import { publicRequest } from 'requests'
import { useQuery } from 'react-query'

type getProfileInfoType = () => Promise<ProfileInfo>

export const getProfileInfo: getProfileInfoType = async () => {
  try {
    const response = await publicRequest.get(`/profile`)
    return response.data
  } catch (err) {
    return []
  }
}

export const useProfileInfo = () => {
  return useQuery<ProfileInfo, Error>(['profile'], {
    queryFn: () => getProfileInfo(),
    refetchOnWindowFocus: false,
  })
}
