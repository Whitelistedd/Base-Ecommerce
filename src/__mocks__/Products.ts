import { TEST_PRODUCT1, TEST_PRODUCT2, TEST_PRODUCT3 } from './Product'

const TEST_PRODUCTS = [TEST_PRODUCT1, TEST_PRODUCT2, TEST_PRODUCT3]

export default TEST_PRODUCTS
