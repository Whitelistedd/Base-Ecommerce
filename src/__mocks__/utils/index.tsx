import { Provider } from 'react-redux'
import { QueryClientProvider } from '@tanstack/react-query'
import { UserProvider } from '@auth0/nextjs-auth0'
import { createTestQueryClient } from './reactQuery'
import { mockStore } from './reactRedux'

export function wrapWithAll(children: React.ReactNode) {
  const testQueryClient = createTestQueryClient()
  return (
    <Provider store={mockStore}>
      <QueryClientProvider client={testQueryClient}>
        <UserProvider>{children}</UserProvider>
      </QueryClientProvider>
    </Provider>
  )
}
