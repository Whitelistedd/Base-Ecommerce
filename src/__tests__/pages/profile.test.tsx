import {
  QueryCache,
  QueryClient,
  QueryClientProvider,
} from '@tanstack/react-query'
import { renderWithClient, wrapWithClient } from '../../__mocks__/utils'

import Profile from '@/pages/profile'
import profileInfo from '@/__mocks__/profileInfo'
import renderer from 'react-test-renderer'
import { useProfileInfo } from '@/hooks/useProfileInfo'
import { useUser } from '@auth0/nextjs-auth0'
import userData from '@/__mocks__/userData'

const mockedUseUser = useUser as jest.Mock

jest.mock('@auth0/nextjs-auth0', () => ({
  useUser: jest.fn(),
}))

const mockedUseProfileInfo = useProfileInfo as jest.Mock

jest.mock('@/hooks/useProfileInfo', () => ({
  useProfileInfo: jest.fn(),
}))

describe('Profile page', () => {
  const queryCache = new QueryCache()
  const queryClient = new QueryClient({ queryCache })

  it('fetches user info with correct params', () => {
    mockedUseUser.mockImplementation(() => ({
      user: userData,
    }))
    mockedUseProfileInfo.mockImplementation(() => ({
      data: profileInfo,
    }))
    renderWithClient(queryClient, <Profile />)
  })

  describe('while loading', () => {
    it('renders a loader', () => {
      mockedUseUser.mockImplementation(() => ({
        user: userData,
        isLoading: true,
      }))
      mockedUseProfileInfo.mockImplementation(() => ({
        data: profileInfo,
        error: true,
      }))

      const { getByAltText } = renderWithClient(queryClient, <Profile />)

      expect(getByAltText(/loading.../)).toBeInTheDocument()
    })
  })

  describe('with an error', () => {
    it('renders a error', () => {
      mockedUseUser.mockImplementation(() => ({
        user: userData,
        error: true,
        isLoading: false,
      }))
      mockedUseProfileInfo.mockImplementation(() => ({
        data: profileInfo,
        error: true,
      }))

      const { getByAltText } = renderWithClient(queryClient, <Profile />)

      expect(getByAltText(/Failed/)).toBeInTheDocument()
    })
  })

  it('should match snapshot', () => {
    mockedUseUser.mockImplementation(() => ({
      user: userData,
      error: false,
      isLoading: false,
    }))
    mockedUseProfileInfo.mockImplementation(() => ({
      error: false,
      data: profileInfo,
    }))
    const tree = renderer.create(wrapWithClient(<Profile />)).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
