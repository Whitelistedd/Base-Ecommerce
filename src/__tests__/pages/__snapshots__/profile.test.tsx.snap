// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile page should match snapshot 1`] = `
<div
  className="sc-4173a692-8 iSLfYC"
>
  <div
    className="sc-2772eec5-4 chSZwi"
  >
    <div
      className="sc-2772eec5-3 deXkCI"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-19w203a-MuiSvgIcon-root"
        data-testid="CloseIcon"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <h1
        className="sc-2772eec5-2 dALrwq"
      >
        EDIT AN ADDRESS
      </h1>
      <p
        className="sc-2772eec5-1 jwYcOp"
      >
        Please fill in the information below:
      </p>
      <form
        className="sc-2772eec5-0 kVSGdo"
        onSubmit={[Function]}
      >
        <div
          className="sc-9417a80c-0"
        >
          <div
            className="MuiFormControl-root MuiTextField-root sc-9417a80c-2 cEDZcC css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            formerror=""
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r0:"
                name="firstName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value="test"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-ihdtdm"
                >
                  <span
                    className="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="sc-9417a80c-0"
        >
          <div
            className="MuiFormControl-root MuiTextField-root sc-9417a80c-2 cEDZcC css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            formerror=""
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r1:"
                name="lastName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value="test"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-ihdtdm"
                >
                  <span
                    className="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="sc-9417a80c-0"
        >
          <div
            className="MuiFormControl-root MuiTextField-root sc-9417a80c-2 cEDZcC css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            formerror=""
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r2:"
                name="address.address"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value="testAddress"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-ihdtdm"
                >
                  <span
                    className="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="sc-9417a80c-0"
        >
          <div
            className="MuiFormControl-root MuiTextField-root sc-9417a80c-2 cEDZcC css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            formerror=""
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r3:"
                name="address.city"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value="testCity"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-ihdtdm"
                >
                  <span
                    className="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="sc-4c80024d-6 jyHSlt"
        >
          <div
            className="sc-ce6427f1-0 hiAETv"
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary sc-ce6427f1-2 jWJUfY css-1fsbs06-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              control={
                {
                  "_defaultValues": {},
                  "_executeSchema": [Function],
                  "_fields": {
                    "address": {
                      "address": {
                        "_f": {
                          "mount": true,
                          "name": "address.address",
                          "ref": {
                            "name": "address.address",
                          },
                          "required": "Enter an address",
                          "value": "testAddress",
                        },
                      },
                      "city": {
                        "_f": {
                          "mount": true,
                          "name": "address.city",
                          "ref": {
                            "name": "address.city",
                          },
                          "required": "Enter a City",
                          "value": "testCity",
                        },
                      },
                      "country": {
                        "_f": {
                          "mount": true,
                          "name": "address.country",
                          "ref": {
                            "name": "address.country",
                          },
                          "required": "Select a Region/Country",
                          "value": "",
                        },
                      },
                      "zipCode": {
                        "_f": {
                          "mount": true,
                          "name": "address.zipCode",
                          "ref": {
                            "name": "address.zipCode",
                          },
                          "required": "Enter a postal code",
                          "value": "testZipCode",
                        },
                      },
                    },
                    "firstName": {
                      "_f": {
                        "mount": true,
                        "name": "firstName",
                        "ref": {
                          "name": "firstName",
                        },
                        "required": "Enter a First name",
                        "value": "test",
                      },
                    },
                    "lastName": {
                      "_f": {
                        "mount": true,
                        "name": "lastName",
                        "ref": {
                          "name": "lastName",
                        },
                        "required": "Enter a Last name",
                        "value": "test",
                      },
                    },
                    "phoneNumber": {
                      "_f": {
                        "mount": true,
                        "name": "phoneNumber",
                        "ref": {
                          "name": "phoneNumber",
                        },
                        "required": "Enter a phone number",
                        "value": undefined,
                      },
                    },
                  },
                  "_focusError": [Function],
                  "_formState": {
                    "dirtyFields": {},
                    "errors": {},
                    "isDirty": false,
                    "isSubmitSuccessful": false,
                    "isSubmitted": false,
                    "isSubmitting": false,
                    "isValid": false,
                    "isValidating": false,
                    "submitCount": 0,
                    "touchedFields": {},
                  },
                  "_formValues": {
                    "address": {
                      "address": "testAddress",
                      "city": "testCity",
                      "country": "",
                      "zipCode": "testZipCode",
                    },
                    "firstName": "test",
                    "lastName": "test",
                    "phoneNumber": undefined,
                  },
                  "_getDirty": [Function],
                  "_getFieldArray": [Function],
                  "_getWatch": [Function],
                  "_names": {
                    "array": Set {},
                    "mount": Set {
                      "firstName",
                      "lastName",
                      "address.address",
                      "address.city",
                      "address.country",
                      "address.zipCode",
                      "phoneNumber",
                    },
                    "unMount": Set {},
                    "watch": Set {},
                  },
                  "_options": {
                    "mode": "onSubmit",
                    "reValidateMode": "onChange",
                    "shouldFocusError": true,
                  },
                  "_proxyFormState": {
                    "dirtyFields": false,
                    "errors": "all",
                    "isDirty": false,
                    "isValid": false,
                    "isValidating": false,
                    "touchedFields": false,
                  },
                  "_removeUnmounted": [Function],
                  "_stateFlags": {
                    "action": false,
                    "mount": false,
                    "watch": false,
                  },
                  "_subjects": {
                    "array": {
                      "next": [Function],
                      "observers": [],
                      "subscribe": [Function],
                      "unsubscribe": [Function],
                    },
                    "state": {
                      "next": [Function],
                      "observers": [],
                      "subscribe": [Function],
                      "unsubscribe": [Function],
                    },
                    "watch": {
                      "next": [Function],
                      "observers": [],
                      "subscribe": [Function],
                      "unsubscribe": [Function],
                    },
                  },
                  "_updateFieldArray": [Function],
                  "_updateValid": [Function],
                  "getFieldState": [Function],
                  "register": [Function],
                  "unregister": [Function],
                }
              }
              isClearable={true}
              onClick={[Function]}
              required="Select a Region/Country"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-address.country"
                className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-component-select-address.country"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  className="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden={true}
                className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="address.country"
                onAnimationStart={[Function]}
                onChange={[Function]}
                required="Select a Region/Country"
                tabIndex={-1}
                value=""
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-ihdtdm"
                >
                  <span
                    className="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="sc-9417a80c-0"
          >
            <div
              className="MuiFormControl-root MuiTextField-root sc-9417a80c-2 cEDZcC css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              formerror=""
            >
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled={false}
                  id=":r4:"
                  name="address.zipCode"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value="testZipCode"
                />
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-ihdtdm"
                  >
                    <span
                      className="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-9417a80c-0"
        >
          <div
            className="MuiFormControl-root MuiTextField-root sc-9417a80c-2 cEDZcC css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            formerror=""
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r5:"
                name="phoneNumber"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-ihdtdm"
                >
                  <span
                    className="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium sc-d4e95e10-0 cscvRh css-1wv28lq-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          data-testid="Button"
          disabled={false}
          id=":r6:"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          EDIT AN ADDRESS
        </button>
      </form>
    </div>
  </div>
  <div
    className="sc-4173a692-0 cOvppE"
  >
    <div
      className="sc-6e310889-1"
    >
      <a
        href="/profile"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <p
          className="sc-6e310889-0 hzQdTS"
        >
          GO BACK TO PROFILE
        </p>
      </a>
    </div>
    <p
      className="sc-4173a692-6"
    >
      Welcome back, 
      user
    </p>
    <div
      className="sc-4173a692-4 hWpjpZ"
    >
      <div
        className="sc-4173a692-1 bzJlkL"
      >
        <p
          className="sc-4173a692-5 fvNEVX"
        >
          MY ORDERS
        </p>
        <p
          className="sc-4173a692-6"
        >
          You havent placed any orders yet
        </p>
      </div>
      <div
        className="sc-4173a692-2 iEqouk"
      >
        <p
          className="sc-4173a692-5 fvNEVX"
        >
          PRIMARY ADDRESS
        </p>
        <div
          className="sc-4173a692-3 bRsCQl"
        >
          <p
            className="sc-4173a692-6"
          >
            First name: 
            test
          </p>
          <p
            className="sc-4173a692-6"
          >
            Last name: 
            test
          </p>
          <p
            className="sc-4173a692-6"
          >
            Address 1: 
            testAddress
          </p>
          <p
            className="sc-4173a692-6"
          >
            City: 
            testCity
          </p>
          <p
            className="sc-4173a692-6"
          >
            Apartment: 
            testApartment
          </p>
          <p
            className="sc-4173a692-6"
          >
            ZIP code: 
            testZipCode
          </p>
        </div>
        <a
          className="sc-4173a692-7 kBBofw"
          onClick={[Function]}
        >
          EDIT ADDRESSES
        </a>
      </div>
    </div>
  </div>
</div>
`;
